tempto-core/src/main/java/io/prestosql/tempto/internal/fulfillment/resources/SuiteResourcesState.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tempto.internal.fulfillment.resources;
+
+import static io.prestosql.tempto.context.ThreadLocalTestContextHolder.testContext;
+
+public class SuiteResourcesState
+        extends ResourcesState
+{
+    public static <T extends AutoCloseable> T closeAfterSuite(T resource)
```

Comment:
End of suite can be many minutes later, so it can be viewed as "controlled leak".
Maybe it could be possible to provide required resources as part of environment setup or test execution setup?

Reply:
why not simply copy `T closeAfterClass(T resource)` from `AbstractTestQueryFramework` to `ProductTest`?

`@AfterTest/@AfterClass` is rather widely understood, `RequirementsFulfiller` is not understood by most people in the company, and has the delay problem @findepi mentioned

Reply:
> Maybe it could be possible to provide required resources as part of environment setup or test execution setup?

@findepi That should be the way to go. `closeAfterSuite` is for resources like `tables` and here you might have immutable like table resources that you want to clean after all tests.

> AbstractTestQueryFramework

@iirekm This class do not exists in tempto.

Reply:
That's why I wrote *copy*, you need to copy this method from AbstractTestQueryFramework from presto-testing .

Reply:
This is the "copy" in a tempto way.